src/mongo/delegates/MongoTripodTables.class.php

```diff
@@ -378,6 +407,310 @@ public function generateTableRows($tableType,$resource=null,$context=null)
+     */
+    protected function doComputedFields(array $spec, array &$dest)
+    {
+        if (isset($spec['computed_fields']))
+        {
+            foreach ($spec['computed_fields'] as $f)
+            {
+                if(isset($f['fieldName']) && isset($f['value']) && is_array($f['value']))
+                {
+                    $computedFunctions = array_intersect_key(self::$computedFieldFunctions, $f['value']);
+                    $dest[$f['fieldName']] = $this->getComputedValue($computedFunctions[0], $f['value'], $dest);
+                }
+            }
+        }
+    }
+
+    /**
+     * @param string $function A defined computed value function
+     * @param array $spec The computed field spec
+     * @param array $dest The table row document to save
+     * @return mixed The computed value
+     */
+    protected function getComputedValue($function, array $spec, array &$dest)
+    {
+        $value = null;
+        switch($function)
+        {
+            case '_conditional_':
+                $value = $this->generateConditionalValue($spec[$function], $dest);
+                break;
+            case '_replace_':
+                $value = $this->generateReplaceValue($spec[$function], $dest);
+                break;
+            case '_arithmetic_':
+                $value = $this->computeArithmeticValue($spec[$function], $dest);
+                break;
+        }
+
+        return $value;
+    }
+
+    /**
+     * @param array $equation
+     * @param array $dest
+     * @return float|int|null
+     * @throws InvalidArgumentException
+     */
+    protected function computeArithmeticValue(array $equation, array &$dest)
+    {
+        if(count($equation) < 3)
+        {
+            throw new InvalidArgumentException("Equations must consist of an array with 3 values");
+        }
+        if(!in_array($equation[1], self::$arithmeticOperators))
+        {
+            throw new InvalidArgumentException("Invalid arithmetic operator");
+        }
+
+        $left = $this->rewriteVariableValue($equation[0], $dest, 'numeric');
+        $right = $this->rewriteVariableValue($equation[2], $dest, 'numeric');
+        if(is_array($left))
+        {
+            $left = $this->computeArithmeticValue($left, $dest);
+        }
+        if(is_array($right))
+        {
+            $right = $this->computeArithmeticValue($right, $dest);
+        }
+
+        switch($equation[1])
+        {
+            case "+":
+                $value = $left + $right;
+                break;
+            case "-":
+                $value = $left - $right;
+                break;
+            case "*":
+                $value = $left * $right;
+                break;
+            case "/":
+                $value = $left / $right;
+                break;
+            case "%":
+                $value = $left % $right;
+                break;
+            default:
+                $value = null;
+        }
```

Comment:
No - it should have been removed from the valid operators: exponents were introduced in PHP 5.6, so I don't think we should support them here.